### ISSUE TITLE
22, invalid mode ('r') or filename ""

### ISSUE THREAD

**Issue by CoffeeProcessingUnit** [bitcraft/PyTMX#123]
A few days ago, I tried launching a pygame program I've written a couple of years ago. Since I got a new computer recently, I reinstalled everything. When I tried to run the game I ran into a few issues, which I managed to resolve quickly, but now I'm stuck. 

I learned that the error mentioned above often occurs when a file path is misinterpreted, when slashes are used the wrong way. However, this doesn't seem to be the problem here, since I've tried all kinds of variations. Furthermore, as you can see, the invalid filename seems to be an empty string for whatever reason. 

The error occurs when I use the load_pygame(filename) function. The filename used in the function call is definitely not the problem. 

Here is what Python is telling me (I hope someone can make sense of it, because I can't):

`File "C:\Python27\lib\site-packages\pytmx\util_pygame.py", line 141, in load_pygame
    return pytmx.TiledMap(filename, *args, **kwargs)
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 360, in __init__
    self.parse_xml(ElementTree.parse(self.filename).getroot())
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 397, in parse_xml
    self.add_layer(TiledObjectGroup(self, subnode))
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 1096, in __init__
    self.parse_xml(node)
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 1106, in parse_xml
    for child in node.findall('object'))
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 1106, in <genexpr>
    for child in node.findall('object'))
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 1134, in __init__
    self.parse_xml(node)
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 1154, in parse_xml
    self._set_properties(node)
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 273, in _set_properties
    properties = parse_properties(node)
  File "C:\Python27\lib\site-packages\pytmx\pytmx.py", line 212, in parse_properties
    d[subnode.get('name')] = cls(subnode.get('value')) if cls is not None else subnode.get('value')
IOError: [Errno 22] invalid mode ('r') or filename: ''`

Thanks in advance for any help!

**Comment by bitcraft** [comment]
might be a python 2 regression.  im not going to actively support python 2.7 fixes moving forward.... could you try `pip install pytmx==3.21.7` and see if that helps?

**Comment by CoffeeProcessingUnit** [comment]
Thanks for the quick reply!

The version you suggested is already installed. Maybe a previous version would fix it?

**Comment by bitcraft** [comment]
yeah, you could try `pip install pytmx==3.21.6`.  I won't have time to look at this until later today though.  hope that helps.

**Comment by CoffeeProcessingUnit** [comment]
Hi again! Installing 3.21.6 didn't change anything. But since I wrote the code a few years ago, I tried the last release from 2016 (3.20.16), and it works just fine. I'm not going to continue working on the project. I just wanted to see the code in action once again, in order to see if I could use some of the logic for my current project. So as far as I'm concerned, this fixes the problem. Thanks again for your help!
